### DISCUSSION TITLE
wavesurfer.backend undefined

### DISCUSSION THREAD

**Discussion by Javarome** [katspaugh/wavesurfer.js#3951] [Q&A]
Hello, I'm trying to migrate from wavesurfer 6 to 7 and when receiving the "ready" event, wavesurfer.backend is not defined. How is it possible?
Here are the options I use:
```
{
autoCenter:  true
autoScroll: true
autoplay: false
backend: "MediaElement"
barGap: null
barHeight: 0.9
barWidth: 1
container: div#waveform
cursorWidth: 1
dragToSeek: false
fetchParams: {headers: null}
fillParent: true
height: 150
interact: true
minPxPerSec: 0
plugins: (2) [_r, _o]
progressColor: CanvasGradient {}
responsive: 500
sampleRate: 8000
smartRender: true
splitChannels: true
splitChannelsOptions: {overlay: true}
waveColor: "#EBEDF4"
}
```
I tried to understand it from the sources, but I can't see even any backend assignment (`xxx.backend = something`) in the code.
So I'm wondering if this property has been removed in waveform 7, and what would be the replacement for code like `wavesurfer.backend.ac.xxx` for instance.

Thanks for your help

**Comment by katspaugh** [comment]
`backend` isn't a thing anymore (it was never part of the official API). See the details here: https://github.com/katspaugh/wavesurfer.js?tab=readme-ov-file#upgrading-from-v6

If you need access to AudioContext, the recommended approach is to initialize it yourself like in this example: https://wavesurfer.xyz/examples/?webaudio.js as it's not the default playback engine anymore.
